tools/websoc-scraper-v2/index.ts

```diff
@@ -0,0 +1,453 @@
+                  waitlistFull: section.status === "FULL",
+                  overEnrolled:
+                    parseInt(section.numCurrentlyEnrolled.totalEnrolled) >
+                    parseInt(section.maxCapacity),
+                  cancelled: section.sectionComment.includes(
+                    "***  CANCELLED  ***"
+                  ),
+                  data: isolateSection({ school, department, course, section }),
+                },
+              };
+            }
+          }
+        }
+      }
+    }
+    for (const [geCategory, response] of Object.entries(data.ge)) {
+      for (const school of response.schools) {
+        for (const department of school.departments) {
+          for (const course of department.courses) {
+            for (const section of course.sections) {
+              if (res[`${term} ${section.sectionCode}`]) {
+                res[`${term} ${section.sectionCode}`].data.geCategories.push(
+                  geCategory as GE
+                );
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  logger.info(`Processed ${Object.keys(res).length} sections`);
+  const sectionsCreated = await prisma.websocSection.createMany({
+    data: Object.values(res).map((d) => d.data),
+  });
+  logger.info(`Inserted ${sectionsCreated.count} sections`);
+  const instructorsCreated = await prisma.websocSectionInstructor.createMany({
+    data: Object.values(res).flatMap((d) => d.meta.instructors),
+  });
+  logger.info(`Inserted ${instructorsCreated.count} instructors`);
+  const meetingsCreated = await prisma.websocSectionMeeting.createMany({
+    data: Object.values(res).flatMap((d) => d.meta.meetings),
+  });
+  logger.info(`Inserted ${meetingsCreated.count} meetings`);
+  const instructorsDeleted = await prisma.websocSectionInstructor.deleteMany({
+    where: {
+      year: term.year,
+      quarter: term.quarter,
+      timestamp: { lt: timestamp },
+    },
+  });
+  logger.info(`Removed ${instructorsDeleted.count} instructors`);
+  const meetingsDeleted = await prisma.websocSectionMeeting.deleteMany({
+    where: {
+      year: term.year,
+      quarter: term.quarter,
+      timestamp: { lt: timestamp },
+    },
+  });
+  logger.info(`Removed ${meetingsDeleted.count} meetings`);
+  const sectionsDeleted = await prisma.websocSection.deleteMany({
+    where: {
+      year: term.year,
+      quarter: term.quarter,
+      timestamp: { lt: timestamp },
+    },
+  });
+  logger.info(`Removed ${sectionsDeleted.count} sections`);
+  logger.info("Sleeping for 5 minutes");
+  await sleep(SLEEP_DURATION);
+}
+
+/**
+ * The entry point of the program.
+ */
+(async function main() {
+  try {
+    logger.info("websoc-scraper-v2 daemon starting");
+    let now = new Date();
+    let termsInScope = await getTermsToScrape(now);
+    for (const [name, term] of Object.entries(termsInScope))
+      await scrape(name, term);
+    for (;;) {
```

Comment:
```suggestion
    while (true) {
```
Idk, I just don't like `for (;;)` lol.

I think it could be worth looking into using pm2's auto-restart with delay as opposed to an infinite loop. We'd be able to manage the instance and adjust the delay, restart conditions, etc. more easily. Encapsulating the entire program as an infinite loop just doesn't feel like the best long-term solution.

Reply:
for(;;) is cursed

Reply:
ESLint doesn't like constants in while loop conditions. That alone is probably a sign to refactor this into a script that runs only once and have pm2 manage the restart conditions.